internal/backend/storage/fossilfs/fossil.go

```diff
@@ -251,6 +266,26 @@ func (f *Fossil) Commit(ctx context.Context, msg string) error {
 	)
 }
 
+// TryCommit calls commit and returns nil if there was nothing to commit or if the Fossil repo was not initialized.
+func (f *Fossil) TryCommit(ctx context.Context, msg string) error {
+	err := f.Commit(ctx, msg)
+	if err == nil {
+		return nil
+	}
+	if err == store.ErrGitNothingToCommit {
+		debug.Log("Nothing to commit. Ignoring.")
+
+		return nil
+	}
+	if err == store.ErrGitNotInit {
+		debug.Log("Git not initialized. Ignoring.")
```

Comment:
Same as below, not sure how Fossil errors match our `Git` errors?
```suggestion
	if err == store.ErrGitNothingToCommit {
		debug.Log("Nothing to commit. Ignoring.")

		return nil
	}
	if err == store.ErrGitNotInit {
		debug.Log("Fossil not initialized. Ignoring.")
```